### DISCUSSION TITLE
How to call methods of different types in two files？

### DISCUSSION THREAD

**Discussion by luckyboy-wei** [seleniumbase/SeleniumBase#2448] [Q&A] [answered]
For example:

```python
# aa.py

from seleniumbase import BaseCase
BaseCase.main(__name__, __file__)

class EapCase(BaseCase):
    def test_add_eap(self):
        self.open("http://192.168.11.11:80/")
```

```python
# bb.py

import aa
from seleniumbase import BaseCase
BaseCase.main(__name__, __file__)

class FdcCase(BaseCase):
    def test_add_fdc(self):
        aa.EapCase().test_add_eap()  # Call failed here

        self.open("http://192.168.11.11:80/")
```

I want to call the method from `bb` in `aa`.

`aa.EapCase().test_add_eap()`  # Call failed here

**Comment by mdmintz** [comment, marked as answer]
The example from [SeleniumBase/examples/boilerplates/classic_obj_test.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/boilerplates/classic_obj_test.py) covers your scenario:

```python
"""Classic Page Object Model with BaseCase inheritance."""
from seleniumbase import BaseCase
BaseCase.main(__name__, __file__)

class DataPage:
    def go_to_data_url(self, sb):
        sb.open("data:text/html,<p>Hello!</p><input />")

    def add_input_text(self, sb, text):
        sb.type("input", text)

class ObjTests(BaseCase):
    def test_data_url_page(self):
        DataPage().go_to_data_url(self)
        self.assert_text("Hello!", "p")
        DataPage().add_input_text(self, "Goodbye!")
```

After you modify your files to correctly use that format, they will look like this:

```python
# aa.py

class EapCase:
    def test_add_eap(self, sb):
        sb.open("http://192.168.11.11:80/")
```

```python
# bb.py
import aa

class FdcCase(BaseCase):
    def test_add_fdc(self):
        aa.EapCase().test_add_eap(self)

        self.open("http://192.168.11.11:80/")
```

**Reply by luckyboy-wei** [reply]
I don't want to use it

sb.open()
This format

**Comment by luckyboy-wei** [comment]
![66](https://github.com/seleniumbase/SeleniumBase/assets/49641020/2f9c0120-a036-437d-a78b-1bf40c76241e)


Can I keep this format on both sides? Is there any other solution

Thank you

**Reply by mdmintz** [reply]
The other alternative involves subclassing. (See: [SeleniumBase/examples/boilerplates/base_test_case.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/boilerplates/base_test_case.py))

In your case, that would mean that your `BaiduCase` class would have to directly inherit `GoogleCase` instead of `BaseCase`.
